common4j/src/main/com/microsoft/identity/common/java/cache/IAccountCredentialCache.java

```diff
@@ -160,6 +160,31 @@ List<Credential> getCredentialsFilteredBy(
             final String requestedClaims
     );
 
+    /**
+     * Returns all of the Credentials matching the supplied criteria from the input list.
+     *
+     * @param homeAccountId    The homeAccountId used to match Credential cache keys.
+     * @param environment      The environment used to match Credential cache keys.
+     * @param credentialType   The sought CredentialType.
+     * @param clientId         The clientId used to match Credential cache keys.
+     * @param realm            The realm used to match Credential cache keys.
+     * @param target           The target used to match Credential cache keys.
+     * @param requestedClaims  The requested claims used to match Credential cache keys.
+     * @param inputCredentials The list of credentials to be filtered from.
+     * @return A mutable List of Credentials matching the supplied criteria.
+     */
+    List<Credential> getCredentialsFilteredBy(
+            final String homeAccountId,
+            final String environment,
+            final CredentialType credentialType,
```

Comment:
These should be marked as Nullable if we do allow them to be set to null.

Reply:
This is marked on the concrete implementation, but not the interface definition. 

Either
- The subclass dictates the behavior of handling or not handling `null` and the interface has no opinion (current)

Or
- The interface defines how subclasses should be handling or not handling `null` (alternative option)

Re-reviewing my own code here, option 2 looks better from a static analysis and Liskov substitution standpoint